src/test/spec/write_error.rs

```diff
@@ -12,8 +12,9 @@ async fn details() {
     let _guard = LOCK.run_concurrently().await;
     let client = EventClient::new().await;
 
-    // TODO: RUST-894 unskip once SERVER-58399 is fixed.
-    if client.server_version_lt(5, 0) || client.is_sharded() {
+    if client.server_version_lt(5, 0) {
+        // SERVER-58399
```

Comment:
I think we can remove this ticket reference now that the issue is resolved

Reply:
I'd prefer to leave it there as a breadcrumb for why servers <5.0 can't run the test.